tests/test_eventgrid.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime
+import unittest
+from typing import List
+
+import azure.functions as func
+import azure.functions.eventgrid as azf_event_grid
+
+
+class MyTestCase(unittest.TestCase):
+    def test_eventgrid_input_type(self):
+        check_input_type = azf_event_grid.EventGridEventInConverter.\
+            check_input_type_annotation
+        self.assertTrue(check_input_type(func.EventGridEvent))
+        self.assertFalse(check_input_type(List[func.EventGridEvent]))
+        self.assertFalse(check_input_type(str))
+        self.assertFalse(check_input_type(bytes))
+
+    def test_eventgrid_output_type(self):
+        check_output_type = azf_event_grid.EventGridOutConverter.\
+            check_output_type_annotation
+        self.assertTrue(check_output_type(func.EventGridEvent))
+        self.assertTrue(check_output_type(List[func.EventGridEvent]))
+        self.assertTrue(check_output_type(str))
+        self.assertTrue(check_output_type(bytes))
+        self.assertTrue(check_output_type(List[str]))
+
+    def test_eventgrid_decode(self):
+        eventGridEvent = azf_event_grid.EventGridEventInConverter.decode(
+            data=self._generate_single_eventgrid_datum(), trigger_metadata=None
+        )
+        self.assertEqual(
+            eventGridEvent.id,
+            "00010001-0001-0001-0001-000100010001")
+        self.assertEqual(eventGridEvent.subject, "eventhubs/test")
+        self.assertEqual(eventGridEvent.event_type, "captureFileCreated")
+        self.assertEqual(eventGridEvent.topic, "/TestTopic/namespaces/test")
+        self.assertIsNotNone(eventGridEvent.get_json())
+
+    def test_eventgrid_decode_with_null_data(self):
+        eventGridEvent = azf_event_grid.EventGridEventInConverter.decode(
+            data=self._generate_single_eventgrid_datum(
+                with_data=False), trigger_metadata=None)
+        self.assertEqual(
+            eventGridEvent.id,
+            "00010001-0001-0001-0001-000100010001")
+        self.assertEqual(eventGridEvent.subject, "eventhubs/test")
+        self.assertEqual(eventGridEvent.event_type, "captureFileCreated")
+        self.assertEqual(eventGridEvent.topic, "/TestTopic/namespaces/test")
+        self.assertIsNone(eventGridEvent.get_json())
+
+    def test_eventgrid_encode_with_str_data(self):
+        example_data = self._generate_single_eventgrid_str()
+        eventGridDatum = azf_event_grid.EventGridOutConverter.encode(
+            example_data, expected_type=type(example_data))
+        self.assertEqual(eventGridDatum.type, "string")
+
+    def test_eventgrid_encode_with_bytes_data(self):
+        example_data = self._generate_single_eventgrid_str(True)
+        eventGridDatum = azf_event_grid.EventGridOutConverter.encode(
+            example_data, expected_type=type(example_data))
+        self.assertEqual(eventGridDatum.type, "bytes")
+
+    def test_eventgrid_encode_with_EventGridData(self):
```

Comment:
Is it possible to add an encode List[EventGrid] test case here? If it is not tedious, please add one.

Reply:
Done.